src/logic/notifications/notificationBuilder.tsx

```diff
@@ -261,3 +263,36 @@ export const enhanceSnackbarForAction = (
     ),
   },
 })
+
+export const createTxNotifications = (
+  notifiedTransaction: string,
+  origin: string | null,
+  dispatch: Dispatch,
+): {
+  closePending: () => void
+  showOnError: (err: Error & { code: number }, contractErrorMessage: string | null) => void
+} => {
+  // Notifications
+  // Each tx gets a slot in the global snackbar queue
+  // When multiple snackbars are shown, it will re-use the same slot for
+  // notifications about different states of the tx
+  const notificationSlot = getNotificationsFromTxType(notifiedTransaction, origin)
+  const beforeExecutionKey = dispatch(enqueueSnackbar(notificationSlot.beforeExecution))
+
+  return {
+    closePending: () => dispatch(closeSnackbarAction({ key: beforeExecutionKey })),
+
+    showOnError: (err: Error & { code: number }, contractErrorMessage: string | null) => {
```

Comment:
We should rename the second argument to be more generic as it is no longer always an error from the contract.